apps/nextjs/src/components/Onboarding/AcceptTermsForm.tsx

```diff
@@ -0,0 +1,138 @@
+"use client";
+
+import { useState } from "react";
+
+import { useUser } from "@clerk/nextjs";
+import logger from "@oakai/logger/browser";
+import {
+  OakBox,
+  OakFlex,
+  OakHeading,
+  OakLink,
+  OakP,
+  OakPrimaryButton,
+  OakSpan,
+} from "@oaknational/oak-components";
+import { useReloadSession } from "hooks/useReloadSession";
+import Link from "next/link";
+
+import Button from "@/components/Button";
+import CheckBox from "@/components/CheckBox";
+import SignUpSignInLayout from "@/components/SignUpSignInLayout";
+import TermsContent from "@/components/TermsContent";
+import { trpc } from "@/utils/trpc";
+
+export const AcceptTermsForm = () => {
+  const [dropDownOpen, setDropDownOpen] = useState(true);
+  const { isLoaded } = useUser();
+  const reloadSession = useReloadSession();
+
+  const [termsAcceptedLocal, setTermsAcceptedLocal] = useState(false);
+  const [privacyAcceptedLocal, setPrivacyAcceptedLocal] = useState(false);
+  const setDemoStatus = trpc.auth.setDemoStatus.useMutation();
+  const acceptTerms = trpc.auth.acceptTerms.useMutation({});
+
+  const handleAcceptTermsOfUse = async () => {
+    try {
+      await setDemoStatus.mutateAsync();
+      logger.debug("Demo status set successfully");
+
+      const response = await acceptTerms.mutateAsync({
+        termsOfUse: new Date(),
+        privacyPolicy: privacyAcceptedLocal ? new Date() : false,
+      });
+
+      if (!response?.acceptedTermsOfUse) {
+        throw new Error("Could not accept terms of use");
+      }
+      logger.debug("Terms of use accepted successfully.");
+
+      await reloadSession();
+      logger.debug("Session token refreshed successfully. Redirecting");
+
+      window.location.href = "/";
```

Comment:
Again here, would a ?redirect=<reason> be helpful at all?

Reply:
Done